eidas_node/proxy_service/views.py

```diff
@@ -1 +1,257 @@
+        """
+        Create a light storage instance.
+
+        :param backend: A fully qualified name of the backend class.
+        :param options: The options to pass to the backend.
+        :return: A light storage instance.
+        """
+        return import_from_module(backend)(**options)
+
+    def get_light_request(self) -> LightRequest:
+        """
+        Get a light request.
+
+        :return: A light request.
+        :raise SecurityError: If the request is not found..
+        """
+        request = self.storage.get_light_request(self.light_token.id)
+        if request is None:
+            raise SecurityError('Request not found in light storage.')
+        return request
+
+    def create_saml_request(self, issuer: str) -> SAMLRequest:
+        """
+        Create a SAML request from a light request.
+
+        :param issuer: Issuer of the SAML request.
+        :return: A SAML request.
+        """
+        # Replace the request id with the token id to find the request later during pairing with the response.
+        self.light_request.id = self.light_token.id
+        # Replace the original issuer with our issuer registered at the Identity Provider.
+        self.light_request.issuer = issuer
+        return SAMLRequest.from_light_request(
+            self.light_request,
+            self.request.build_absolute_uri(reverse('identity-provider-response')),
+            datetime.utcnow())
+
+    def get_context_data(self, **kwargs) -> dict:
+        """Adjust template context data."""
+        context = super().get_context_data(**kwargs)
+        context['error'] = self.error
+        if self.saml_request:
+            context['identity_provider_endpoint'] = PROXY_SERVICE_SETTINGS.identity_provider['endpoint']
+            context['saml_request'] = b64encode(dump_xml(self.saml_request.document)).decode('ascii')
+            context['relay_state'] = self.saml_request.relay_state or ''
+        return context
+
+
+class IdentityProviderResponseView(TemplateView):
+    """
+    Forward ab identity provider's response to a service provider.
+
+    eIDAS Generic Proxy Service expect the identity provider's response as a light response.
+    """
+
+    template_name = 'eidas_node/proxy_service/identity_provider_response.html'
+    error = None  # type: Optional[Error]
+    storage = None  # type: LightStorage
+    saml_response = None  # type: SAMLResponse
+    light_response = None  # type: LightResponse
+    light_request = None  # type: LightRequest
+    light_token = None  # type: LightToken
+    encoded_token = None  # type: str
+
+    def get(self, *args, **kwargs):
+        """Handle a HTTP GET request."""
+        return self.http_method_not_allowed(*args, **kwargs)
+
+    def post(self, request: HttpRequest) -> HttpResponse:
+        """Handle a HTTP POST request."""
+        try:
+            self.saml_response = self.get_saml_response(PROXY_SERVICE_SETTINGS.identity_provider.get('key_file'))
```

Comment:
This is a required setting, so no need for `get()`

Reply:
Actually, this is an optional settings but was marked as required in appsettings.

Reply:
OK